pkg/minio/dws.go

```diff
@@ -76,25 +99,41 @@ func (h objectAPIHandlersWrapper) getUserID(r *http.Request, w http.ResponseWrit
 	return res["uuid"], nil
 }
 
-func (h objectAPIHandlersWrapper) bucketNameIsAvailable(r *http.Request) (bool, error) {
-	funcName := "bucketNameIsAvailable"
-	vars := mux.Vars(r)
-	bucket := vars[VarKeyBucket]
-	if bucket == "" {
-		return false, nil
+func (h objectAPIHandlersWrapper) nodeBucketRequest(r *http.Request, method string, bucketName string) (int, error) {
+	sc := 0
+	u := path.Join(h.nodeHost, nodeBucketPath)
+	var reader io.Reader
+	switch method {
+	case "HEAD", "DELETE":
+		u = path.Join(u, bucketName)
+	case "POST":
+		type payload struct {
+			Name string `json:"name"`
+		}
+		p := payload{Name: bucketName}
+		ba, err := json.Marshal(p)
+		if err != nil {
+			h.logger.With("error", err).Error("nodeBucketRequest marshal request body")
+			return sc, fmt.Errorf("marshal request body error: %w", err)
+		}
+		reader = bytes.NewBuffer(ba)
+	default:
+		h.logger.With("method", method).Error("nodeBucketRequest wrong method")
+		return sc, fmt.Errorf("wrong http method: %s", method)
 	}
-
-	res, err := h.authClient.CheckBucketIsUnique(r.Context(), bucket, trustedip.GetClientIP(h.trustedIPs, r))
+	req, err := http.NewRequestWithContext(r.Context(), method, u, reader)
 	if err != nil {
-		h.logger.With("error", err, "bucket name", bucket).Error("check bucket is unique response error")
-		return false, fmt.Errorf("function: %s, could not get response from bucket resolver: %w", funcName, err)
+		h.logger.With("error", err).Error("nodeBucketRequest new request")
+		return sc, fmt.Errorf("could not create a request: %w", err)
 	}
-
-	if res.Error != "" {
-		h.logger.With("error", err, "result", res).Error("check bucket is unique result error")
-		return false, fmt.Errorf("function: %s, error: %s", funcName, res.Error)
+	req.Header.Set("Authorization", "Bearer "+h.nodeToken)
```

Comment:
Could you please remove the "Bearer" from the Authorization header?
Node js team had to remove this part of the header cause they couldn't generate swagger for some reason.
```suggestion
	req.Header.Set("Authorization", h.nodeToken)
```